### DISCUSSION TITLE
using the client without a server for integration testing

### DISCUSSION THREAD

**Discussion by Arikael** [prometheus/client_python#885] [Q&A] [answered]
Hi 

I need to integration test a system/application and while doing that collecting metrics about it (latency, throughput etc.).
Since we use gitlab I use the OpenMetrics format for those tests.

Now to make my live easier I don't want to write this report manually (if I don't have to), since I would only need to collect the metrics for a single test run and export it as file so gitlab can read it I thought maybe it's possible to use the client without connecting to a server.

So the client collects my metrics and exports it to a file.

Can I use the client without connecting to a server'

**Comment by csmarchbanks** [comment, marked as answer]
Hello, yes this is supported. You can generate the output using `openmetrics.exposition.generate_latest` and then persist that to a file if you want to specifically use the OpenMetrics format. There is also `write_to_textfile` which will write metrics to a file in an atomic manner suitable for use with the node_exporter textfile collector. Note that this is in the Prometheus text format, not OpenMetrics.
